cfgov/cfgov/urls.py

```diff
@@ -597,14 +597,48 @@ def empty_200_response(request, *args, **kwargs):
 urlpatterns = urlpatterns + category_redirects
 
 if settings.ALLOW_ADMIN_URL:
-    patterns = [
-        re_path(r'^login/$', login_with_lockout, name='cfpb_login'),
+    # If SAML2 auth is enabled, /login will redirect to /saml2/login,
+    # which in turn will redirect to the configured identity provider.
+    if settings.SAML_AUTH:
+        auth_patterns = [
+            re_path(r'saml2/', include('djangosaml2.urls')),
+            re_path(
+                r'^login/$',
+                RedirectView.as_view(
+                    url='/saml2/login/',
+                    query_string=True
+                ),
+                name='cfpb_login'
+            ),
+            re_path(
+                r'^logout/$',
+                RedirectView.as_view(
+                    url='/saml2/logout/',
+                    query_string=True,
+                ),
+                name='logout'
+            ),
+
+            # For backup/admin/emergency auth, a pure Django login is still
+            # provided.
+            re_path(r'^django_login/$', login_with_lockout, name='cfpb_login'),
```

Comment:
Another possible simplification, that avoids switching which URL goes to which view depending on environment: could we modify [`settings.LOGIN_URL`](https://github.com/cfpb/consumerfinance.gov/blob/bfaf14ad0accbb350538066a6f29b3c1524d28eb/cfgov/cfgov/settings/base.py#L407) if SAML is enabled?

Personally I typically login via /admin/, and when we share admin views around, they get shared as /admin/something. I believe Wagtail will redirect to `LOGIN_URL`, not /login/ specifically, as the way for users to login. If we point `LOGIN_URL` to the SAML login URL when enabled, we can leave urls.py a little simpler because we're just adding a new path, while leaving the old URLs alone.

I'm less sure if there's something similar that can be done with logout.

Reply:
Hmm, I'll see how that works. I think I had an issue with just setting `LOGIN_URL` earlier in my testing, but this has been a very start/stop-y iterative process.

Reply:
Correction, [`WAGTAIL_FRONTEND_LOGIN_URL` appears to only apply to page privacy](https://github.com/wagtail/wagtail/blob/main/wagtail/core/wagtail_hooks.py#L17-L41), not to the general admin.

So, I think we're going to need to at least redirect `/admin/login` (where Wagtail lands you) to the SAML2 URL.

Reply:
Because Wagtail doesn't respect `LOGIN_URL`, and because we have a permanent redirect of `/admin/login` to `/login`, I'm inclined to keep `/login` the point that redirects to SSO.

However, maybe instead of continuing to expose our Wagtail login, we simply allow `/django-admin/login` to function, as our emergency admin login point?

Reply:
This is how it's currently implemented in 5636600.